src/core/use-cases/product/product.test.ts

```diff
@@ -1,18 +1,30 @@
-import { right, left } from 'fp-ts/Either'
+import { pipe } from 'fp-ts/function'
+import * as TE from 'fp-ts/TaskEither'
 import { createProduct, SaveProduct } from './create-product'
 
 const product = { name: 'product', price: 10 }
-const saveProduct: SaveProduct = async (product) => { return product }
-const saveProductError = async (): Promise<never> => {
-  throw new Error('Produto Invalido!')
+const saveProduct: SaveProduct = async (product) => {
+  return `Produto cadastrado com sucesso! ${product.name}`
+}
+const saveProductError: SaveProduct = async () => {
+  throw new Error('Database Error!')
 }
 
-it('should create a product', async () => {
-  const newProduct = await createProduct(product)(saveProduct)
-  expect(newProduct).toEqual(right(product))
+it('Deve criar um produto', async () => {
+  return pipe(
+    product,
+    createProduct(saveProduct),
+    TE.map((newProduct) => expect(newProduct)
```

Comment:
Uma outra coisa pra gente deixar no radar, e testar na próxima mentoria: se algo no código for alterado e esse teste deixar de passar, esse teste vai quebrar?

Minha percepção é de que não, pois se retornar um `left`, o `TE.map` é apenas ignorado.
Talvez a gente tenha que usar o `TE.mapLeft` aqui também, dando `throw` em um erro só pra garantir que o teste vai qauebrar se isso aqui retornar um `Left` =)